builtin/credential/cert/path_crls.go

```diff
@@ -188,44 +195,94 @@ func (b *backend) pathCRLWrite(ctx context.Context, req *logical.Request, d *fra
 	if name == "" {
 		return logical.ErrorResponse(`"name" parameter cannot be empty`), nil
 	}
-	crl := d.Get("crl").(string)
+	if crlRaw, ok := d.GetOk("crl"); ok {
+		crl := crlRaw.(string)
+		certList, err := x509.ParseCRL([]byte(crl))
+		if err != nil {
+			return logical.ErrorResponse(fmt.Sprintf("failed to parse CRL: %v", err)), nil
+		}
+		if certList == nil {
+			return logical.ErrorResponse("parsed CRL is nil"), nil
+		}
 
-	certList, err := x509.ParseCRL([]byte(crl))
-	if err != nil {
-		return logical.ErrorResponse(fmt.Sprintf("failed to parse CRL: %v", err)), nil
-	}
-	if certList == nil {
-		return logical.ErrorResponse("parsed CRL is nil"), nil
+		err = b.setCRL(ctx, req.Storage, certList, name, nil)
+		if err != nil {
+			return nil, err
+		}
+	} else if urlRaw, ok := d.GetOk("url"); ok {
+		url := urlRaw.(string)
+		if url == "" {
+			return logical.ErrorResponse("empty CRL url"), nil
+		}
+		_, err := url2.Parse(url)
+		if err != nil {
+			return logical.ErrorResponse("invalid CRL url: %v", err), nil
+		}
+
+		b.crlUpdateMutex.RLock()
+		if crl, ok := b.crls[name]; ok {
```

Comment:
I'm a bit puzzled by this, nothing will use the crl we are populating within this lock and it would be reset anyways by the setCRL within the fetchCRL call

Also note a `make fmt` is needed. Everything else looks good though!

Reply:
If the CRL already existed and it's valid until was in the future, it would be a no-op in fetchCRL. So we wouldn't validate the URL and wouldn't setCRL.

Reply:
Ah this is tricky, I like what you're doing here: re-setting the same URL on the same existing CRL name will allow users to force a re-fetch of the CRL. I like it!

But shouldn't this be a write lock in that case, strictly?

Reply:
Eugh, `fetchCRL` doesn't look at the `crl.CDP.ValidUntil` value, `updateCRLs` does? So right now we will always reload the URL data?

Reply:
In particular, `fetchCRL` takes a new reference,  _not_ the modified value here in `b.crls`.

Reply:
Ah gotcha, now that it's straight to fetchURL there's no conditional.  Removed the if block and had earlier redone the locking.